lnwallet/chancloser/chancloser.go

```diff
@@ -534,16 +545,18 @@ func (c *ChanCloser) ProcessCloseMsg(msg lnwire.Message) ([]lnwire.Message,
 		// wire.
 		localShutdown, err := c.initChanShutdown()
```

Comment:
`initChanShutdown` calls `MarkCoopBroadcasted` which will update the channel's status at the database level. This will make it so that further commitment updates in the link won't work. `MarkCoopBroadcasted` should instead be called when we've flushed and are ready to send `closing_signed`.

Reply:
I think this can be addressed with a new intermediate state, like `chanFlushing`, we go to that after we recv or send shutdown, then afterwards, we continue execution from here. Either way correct that for the new flow, we need to split out some of the logic from `initChanShutdown`.

Reply:
This is how it is in this PR, no?

Reply:
JK It's not. I'll address.